src/main/java/com/aws/greengrass/config/ConfigurationWriter.java

```diff
@@ -264,4 +264,14 @@ private synchronized void truncateTlog() {
     private synchronized void setTruncateRetryCount() {
         retryCount = count.get() + maxCount / 2;
     }
+
+    /**
+     * Immediately truncate the tlog.
+     */
+    public synchronized void truncateNow() {
+        if (truncateQueued.compareAndSet(false, true)) {
+            context.runOnPublishQueueAndWait(this::truncateTlog);
+            logger.atDebug(TRUNCATE_TLOG_EVENT).log("queued immediate truncation");
```

Comment:
should this log be before the truncate call? it will truncate the log and then log the message right?